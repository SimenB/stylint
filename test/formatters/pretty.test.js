'use strict';

const prettyFormatter = require('../../src/formatters/pretty');
const formatterMockUtils = require('../../src/utils/formatterMockUtils');

describe('prettyFormatter', () => {
  it('should have correct output on no message', () => {
    expect(prettyFormatter(formatterMockUtils.generateReport())).toMatchSnapshot();
  });

  it('should include kill message', () => {
    const report = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);

    expect(prettyFormatter(report, {}, true)).toMatchSnapshot();
  });

  it('should include max errors and max warnings', () => {
    const report = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);

    expect(prettyFormatter(report, { maxErrors: 5, maxWarnings: 5 })).toMatchSnapshot();
  });

  it('should skip non-valid max errors and max warnings', () => {
    const negativeErrorReport = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);
    const negativeErrorOptions = { maxErrors: -1, maxWarnings: 5 };

    const missingErrorReport = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);
    const missingErrorOptions = { maxWarnings: 5 };

    const missingWarningReport = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);
    const missingWarningOptions = { maxErrors: 2 };

    expect(prettyFormatter(negativeErrorReport, negativeErrorOptions)).toMatchSnapshot();
    expect(prettyFormatter(missingErrorReport, missingErrorOptions)).toMatchSnapshot();
    expect(prettyFormatter(missingWarningReport, missingWarningOptions)).toMatchSnapshot();
  });

  it('should format warning correctly', () => {
    const report = formatterMockUtils.generateReport([
      formatterMockUtils.generateWarning('some file.styl', 'no-undefined'),
    ]);

    expect(prettyFormatter(report)).toMatchSnapshot();
  });

  it('should format error correctly', () => {
    const report = formatterMockUtils.generateReport([
      formatterMockUtils.generateError('some file.styl', 'no-undefined'),
    ]);

    expect(prettyFormatter(report)).toMatchSnapshot();
  });

  it('should format error and warning correctly', () => {
    const error = formatterMockUtils.generateError('some file.styl', 'no-undefined');
    const warning = formatterMockUtils.generateWarning('some file.styl', 'no-undefined');

    expect(prettyFormatter(formatterMockUtils.generateReport([error, warning]))).toMatchSnapshot();
  });

  it('should format column', () => {
    const error = formatterMockUtils.generateError('some file.styl', 'no-undefined');

    error.messages[0].column = 5;

    expect(prettyFormatter(formatterMockUtils.generateReport([error]))).toMatchSnapshot();
  });

  it('should not group files by default', () => {
    const error1 = formatterMockUtils.generateError('some file.styl', 'no-undefined');
    const error2 = formatterMockUtils.generateError('some file.styl', 'no-undefined');
    const error3 = formatterMockUtils.generateError('some other file.styl', 'no-undefined');
    const report = formatterMockUtils.generateReport([error1, error2, error3]);

    expect(prettyFormatter(report)).toMatchSnapshot();
  });

  it('should group files correctly', () => {
    const error1 = formatterMockUtils.generateError('some file.styl', ['no-undefined', 'no-undefined']);
    const error2 = formatterMockUtils.generateError('some other file.styl', 'no-undefined');
    const options = { groupOutputByFile: true };
    const report = formatterMockUtils.generateReport([error1, error2]);

    expect(prettyFormatter(report, options)).toMatchSnapshot();
  });
});
