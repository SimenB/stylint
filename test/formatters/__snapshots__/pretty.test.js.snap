// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettyFormatter should format column 1`] = `
"[4msome file.styl[24m
1:5 [90mno-undefined[39m [31merror[39m This is not OK

Stylint: 1 Errors.
Stylint: 0 Warnings."
`;

exports[`prettyFormatter should format error and warning correctly 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [31merror[39m This is not OK

[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 1 Errors.
Stylint: 1 Warnings."
`;

exports[`prettyFormatter should format error correctly 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [31merror[39m This is not OK

Stylint: 1 Errors.
Stylint: 0 Warnings."
`;

exports[`prettyFormatter should format warning correctly 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors.
Stylint: 1 Warnings."
`;

exports[`prettyFormatter should group files correctly 1`] = `
"some file.styl
FILE           LINEDATA SEVERITY MESSAGE        RULE        
[4msome file.styl[24m 1        [31merror[39m    This is not OK [90mno-undefined[39m
[4msome file.styl[24m 1        [31merror[39m    This is not OK [90mno-undefined[39m

some other file.styl
FILE                 LINEDATA SEVERITY MESSAGE        RULE        
[4msome other file.styl[24m 1        [31merror[39m    This is not OK [90mno-undefined[39m

Stylint: 3 Errors.
Stylint: 0 Warnings."
`;

exports[`prettyFormatter should have correct output on no message 1`] = `""`;

exports[`prettyFormatter should include kill message 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors.
Stylint: 1 Warnings.
Stylint: Over Error or Warning Limit."
`;

exports[`prettyFormatter should include max errors and max warnings 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors. (Max Errors: 5)
Stylint: 1 Warnings. (Max Warnings: 5)"
`;

exports[`prettyFormatter should not group files by default 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [31merror[39m This is not OK

[4msome file.styl[24m
1 [90mno-undefined[39m [31merror[39m This is not OK

[4msome other file.styl[24m
1 [90mno-undefined[39m [31merror[39m This is not OK

Stylint: 3 Errors.
Stylint: 0 Warnings."
`;

exports[`prettyFormatter should skip non-valid max errors and max warnings 1`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors.
Stylint: 1 Warnings. (Max Warnings: 5)"
`;

exports[`prettyFormatter should skip non-valid max errors and max warnings 2`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors.
Stylint: 1 Warnings. (Max Warnings: 5)"
`;

exports[`prettyFormatter should skip non-valid max errors and max warnings 3`] = `
"[4msome file.styl[24m
1 [90mno-undefined[39m [33mwarning[39m This is not OK

Stylint: 0 Errors. (Max Errors: 2)
Stylint: 1 Warnings."
`;
